homeassistant/components/airly/config_flow.py

```diff
@@ -37,23 +43,24 @@ async def async_step_user(self, user_input=None):
                 f"{user_input[CONF_LATITUDE]}-{user_input[CONF_LONGITUDE]}"
             )
             self._abort_if_unique_id_configured()
-            api_key_valid = await self._test_api_key(websession, user_input["api_key"])
-            if not api_key_valid:
-                self._errors["base"] = "invalid_api_key"
-            else:
-                location_valid = await self._test_location(
+            try:
+                location_valid = await test_location(
                     websession,
                     user_input["api_key"],
                     user_input["latitude"],
                     user_input["longitude"],
                 )
+            except AirlyError as err:
+                if err.status_code == HTTP_UNAUTHORIZED:
+                    self._errors["base"] = "invalid_api_key"
```

Comment:
A future clean should be to not store the errors in an instance attribute. The form errors are local to the step method of the form and should not be shared or stored.

Reply:
Ok, I'll change this.